modernsyntax/lambda.cpp

```diff
@@ -13,7 +13,8 @@ void incrementCounter(int* counter, const unsigned int times){
 int main(){
   int counter(0);
   // launch a thread to increment the counter
-  std::thread increment(incrementCounter,&counter,100000);
+  
+  std::thread increment([](int* counter, const unsigned int times){for (unsigned int i=0; i<times;++i){++(*counter);}},&counter,100000);
```

Comment:
The number of iterations one could as well try to catch by value in the `[]` part of the lambda.

Reply:
Can that syntax catch an argument passed by `std::thread` or would this require a pre-initialisation of a variable to catch? At the very least a variable initialisation adds another line of code where, to my understanding, the current implementation does not sacrifice performance.

Reply:
To put your point to the extreme - if you specify that number in the same line as the lambda, you don't even need it as extra function parameter of the lambda ;-) Thus having an extra variable seems to better match the lambda definition. Nevertheless the current code is correct.

Reply:
Which means the "better" solution depends on the intended purpose of the count. An external variable is "future-proof" for end-user input while a deliberately hardcoded value is easier to track within the function definition. Passing a fixed value as an argument as in the current version would only be desirable if it is reused multiple times within the function, correct? (then again such a function probably should not be an inline lambda to begin with)

Reply:
Nothing to add to that analysis :-)